models/repo.go

```diff
@@ -2481,17 +2481,78 @@ func (repo *Repository) CustomAvatarPath() string {
 	return filepath.Join(setting.RepositoryAvatarUploadPath, repo.Avatar)
 }
 
-// RelAvatarLink returns a relative link to the user's avatar.
-// The link a sub-URL to this site
-// Since Gravatar support not needed here - just check for image path.
+// GenerateRandomAvatar generates a random avatar for repository.
+func (repo *Repository) GenerateRandomAvatar() error {
+	return repo.generateRandomAvatar(x)
+}
+
+func (repo *Repository) generateRandomAvatar(e Engine) error {
+	idToString := fmt.Sprintf("%d", repo.ID)
+
+	seed := idToString
+	img, err := avatar.RandomImage([]byte(seed))
+	if err != nil {
+		return fmt.Errorf("RandomImage: %v", err)
+	}
+
+	repo.Avatar = idToString
+	if err = os.MkdirAll(filepath.Dir(repo.CustomAvatarPath()), os.ModePerm); err != nil {
+		return fmt.Errorf("MkdirAll: %v", err)
+	}
+	fw, err := os.Create(repo.CustomAvatarPath())
+	if err != nil {
+		return fmt.Errorf("Create: %v", err)
+	}
+	defer fw.Close()
+
+	if _, err := e.ID(repo.ID).Cols("avatar").Update(repo); err != nil {
+		return err
+	}
+
+	if err = png.Encode(fw, img); err != nil {
+		return fmt.Errorf("Encode: %v", err)
+	}
+
+	log.Info("New random avatar created for repository: %d", repo.ID)
+	return nil
+}
+
+// RemoveRandomAvatars removes the randomly generated avatars that were created for repositories
+func RemoveRandomAvatars() error {
```

Comment:
Why we need to remove the random avatars?

Reply:
Random avatars are placed on the file system and set in the database.

Let's say an owner used generated repository avatars and now wants to display a default avatar instead. The generated avatars are stored just like uploaded avatars on the file system (the only difference is the file name) and also update the repository in the database.
The default avatar will be used when no image was added to the repository, which is why the randomly generated images will still be used instead of the default avatar.

Reply:
When a repository has a random avatar, delete it from UI will make it as default image and upload a new avatar will delete the previous image. So I think the cron task is unnecessary?

Reply:
True. But still, looking at an organization with 100 repositories an admin would have to through all projects and delete the avatars. That's why I wanted to provide an easy way to do so, which is why I placed it in the administration panel.
It's not supposed to be a cron task that is executed regularly. Maybe it would be better to put that into a cmd task like `gitea clear repository-avatars`...